### PR TITLE
Sync: Migrate old favourites

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/0/1174433894299346/1204733967627383

### Description
See task description

### Steps to test this PR
See task description

### CODE REVIEW

**Comment by malmstein** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3846** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3846?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3846?utm_source=stack-comment).

**Comment by malmstein** [comment]
Discussed offline with @cmonfortep . We won’t improve the migration logic for now, and move on with what we have

File: app/src/main/java/com/duckduckgo/app/bookmarks/migration/AppDatabaseBookmarksMigrationCallback.kt [hunk 1]
```diff
@@ -189,6 +194,32 @@ class AppDatabaseBookmarksMigrationCallback(
...
+                bookmarks.contains(rootFavorite)
+            }
+        }
+    }
+
+    private fun runOldFavouritesMigration(favourites: List<Entity>) {
+        with(appDatabase.get()) {
+            val favouriteMigration = mutableListOf<Relation>()
+            favourites.forEach {
+                val notAsBookmark = syncRelationsDao().relationByEntityId(it.entityId) == null
```
Comment (cmonfortep) [inline app/src/main/java/com/duckduckgo/app/bookmarks/migration/AppDatabaseBookmarksMigrationCallback.kt:214]: we already know is not a bookmark right? why this check?

**Review by cmonfortep — APPROVED** [review]
LGTM, just a comment about the second query inside`runOldFavouritesMigration`

Haven't tested the migrations again, @malmstein confirmed just ensuring test suite green is ok. 👍

**Comment by malmstein** [comment]
Tested and working, thanks @cmonfortep !

**Pull request merged**